python/essex-config/essex_config/sources/file_source.py

```diff
@@ -0,0 +1,34 @@
+"""File source class implementation."""
+
+import json
+from pathlib import Path
+from typing import Any
+
+import tomllib
+
+from essex_config.sources.source import Source
+
+
+class FileSource(Source):
+    """Class to get the configuration from a file."""
+
+    def __init__(self, file_path: Path | str):
+        """Initialize the class."""
+        if isinstance(file_path, str):
+            file_path = Path(file_path)
+        self.file_path = file_path
+
+    def get_data(self) -> dict[str, Any]:
+        """Get the data dictionary."""
+        if self.file_path.suffix == ".json":
```

Comment:
Would be help to add yaml here as a default as well. For example, GraphRAG --init creates a settings.yaml by default

Reply:
Note that yaml (e.g. as used in GraphRAG) does token replacement for env vars

Reply:
+1 to yaml. Not a blocker on this PR, but will like to have yaml compat and eventually migrate GraphRAG into a simpler config approach.

Reply:
I added basic yaml support :)